src/apps/handlers/prepareMovementData.js

```diff
@@ -1,10 +1,23 @@
 export default function getMovementData(data) {
   // eslint-disable-next-line no-unused-vars
-  const speeds = Object.entries(data.system.attributes.movement).filter(([_, speed]) => speed);
+  const hover = data.system.attributes.movement.traits.hover
+  const speeds = Object.entries(data.system.attributes.movement).filter(([key, speed]) => {
+    if(!speed && key === "fly" && hover){
+      return !speed
+    } else {
+      return speed
+    }
```

Comment:
```js
if ( key === "fly" && !speed && hover) { return true; }
if ( key === "traits") { return false; }
return speed
```

This can be simplified to this.

Reply:
To stay consistent with our style, this should probably be:
```js
if ( key === "fly" && !speed && hover) return true;
if ( key === "traits") return false;
return speed;
```
That being said, we can do better than this as well. We don't need to check that there's no speed when checking for hover. The presence of the hover trait is enough to show the fly speed regardless of the fly speed value.
```js
if ( key === "fly" && hover) return true;
if ( key === "traits") return false;
return speed;
```